src/Socket.php

```diff
@@ -756,123 +748,65 @@ public static function showListForItem($item)
         if ($item->isNewID($item->getID())) {
             return false;
         }
+        $rand = mt_rand();
 
        // Link to open a new socket
         if ($item->getID() && self::canCreate()) {
-            echo "<div class='firstbloc'>";
-            echo "\n<form method='get' action='" . Socket::getFormURL() . "'>\n";
-            echo "<input type='hidden' name='_from_items_id' value='" . $item->getID() . "'>\n";
-            echo "<input type='hidden' name='_from_itemtype' value='" . $item->getType() . "'>\n";
-            echo "<div class='firstbloc'><table class='tab_cadre_fixe'>\n";
-            echo "<tr class='tab_bg_2'>";
-            echo "<td class='tab_bg_2 center'>";
-            echo "<td>\n";
-
-            echo "<button class='btn btn-primary' type='submit' name='_add_fromitem'>";
-            echo "<span>" . __('New socket for this item...') . "</span>";
-            echo "</button>";
-
-            echo "</td>";
-            echo "<td>\n";
-            echo __('Add several sockets');
-            echo "&nbsp;<input type='checkbox' name='several' value='1'></td>\n";
-            echo "</tr></table></div>\n";
-            echo "</form></div>";
+            $twig_params = [
+                'item' => new Socket(),
+                'from_item' => $item,
+                'params' => [
+                    'canedit' => true
+                ]
+            ];
+            echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
```

Comment:
The whole rendering of this method should be moved into a dedicated Twig file that
1. render this small form,
2. includes `components/datatable.html.twig` to render the list.